app/workers/pull_request_monitor_handlers/path_based_labeler.rb

```diff
@@ -0,0 +1,27 @@
+class PullRequestMonitorHandlers::PathBasedLabeler
+  include Sidekiq::Worker
+  sidekiq_options :queue => :miq_bot
+
+  include BranchWorkerMixin
+
+  def perform(branch_id)
+    return unless find_branch(branch_id, :pr)
+    return unless verify_branch_enabled
+
+    process_branch
+  end
+
+  private
+
+  def process_branch
+    label_rules.each do |rule|
+      if diff_file_names.any? { |file_name| file_name =~ Regexp.new(rule.pattern) }
+        GithubService.add_labels_to_an_issue(fq_repo_name, pr_number, [rule.label])
```

Comment:
Oh I see...in that case we already had the fetched labels, but here we would have to issue a query to find all of the existing labels.

Maybe we should re-verify the old assumptions above.  Perhaps the GitHub API was corrected to no longer do this.

These two tests should verify:

- `GithubService.add_labels_to_an_issue(fq_repo_name, pr_number, ["foo"])` and see if it creates a new label in the repo or raises an issue.  If it still does the former, we'll have to deal with that potentially (or maybe it's not an issue since this configuration is not user-editable)
- Create an issue, add the bug label, then call: `GithubService.add_labels_to_an_issue(fq_repo_name, pr_number, ["bug"])` and see if it makes a notification in the UI about adding a label even if it already exists.  If it still does that, we will most definitely have to deal with it otherwise everytime someone does a git push, you will get that "added a label" notification